### DISCUSSION TITLE
Verify Email with Auth

### DISCUSSION THREAD

**Discussion by lalanankit** [laravel/framework#34470] [General]
- Laravel Version: 8
- PHP Version: >7
- Database Driver & Version: MySQL

### Description:
Laravel Version 8 with Admin LTE template.

### Current behavior
Unable to send emails for reset password or user registration.

### Expected Behavior
Sending emails and Verified At timestamp populated to users table.

### Changes to register controller
![6A9A4F04-AEF1-4906-9346-58052DA81EAA](https://user-images.githubusercontent.com/71725903/93932937-a2d7b000-fd40-11ea-85a6-f89357eb9fd3.png)

### Changes to user model:

class User extends Authenticatable implements MustVerifyEmail
{
    use HasFactory, Notifiable;

### Changes to routes, web.php
Auth::routes(['verify' => true]);

### Steps To Reproduce:
Create a project with Laravel 8 and auth, configure the email settings, run migration and create a user with Register Link from the home page

There is an entry in database but exception on the app.

Exception:
![429596EE-6E0C-4B8C-B6E3-B8299765278F](https://user-images.githubusercontent.com/71725903/93933006-baaf3400-fd40-11ea-8914-05632a735747.png)

**Comment by hsmfawaz** [comment]
can you provide your mail configuration 
maybe its just a problem with the mail configuration

**Reply by lalanankit** [reply]
Yes, That was a mail configuration issue.

The MAIL_FROM_ADDRESS when made equal to MAIL_USERNAME, that resolved the problem.
Thank you.

**Comment by lalanankit** [comment]
This has been resolved and can be closed.
Solution: 

Make the following changes to .env file:

```
MAIL_MAILER=smtp
MAIL_HOST=xxxxx.xxxx.xxxxx
MAIL_PORT=587
MAIL_USERNAME=hello@ankitlalan.com
MAIL_PASSWORD=MyPassword
MAIL_ENCRYPTION=tls 
MAIL_FROM_ADDRESS=hello@ankitlalan.com
MAIL_FROM_NAME="${APP_NAME}"
```

routes\web.php
change `Auth::routes()` to `Auth::routes(['verify' => true]);`

HomeController.php
Add/edit the constructor
```
public function __construct()
    {
        $this->middleware(['auth', 'verified']);
    } 
```

model\user.php
Update the model to implement the interface

```
use Illuminate\Contracts\Auth\MustVerifyEmail;
use Illuminate\Notifications\Notifiable;

class User extends Authenticatable **implements MustVerifyEmail**
{
    use HasFactory, Notifiable;
..
..
..
}
```

That's it. It will bring the Laravel 8 Auth to your project.
